File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
import sys
sys.setrecursionlimit(10**7)

dp = {
    0 : 0,
    1 : 1,
    2 : 1
}
def solution(n):
    answer = fibo(n)
    return answer %1234567

def fibo(n):
    if n not in dp:
        dp[n] = fibo(n-1) + fibo(n-2)
    return dp[n]

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.swift
import Foundation

var dp: [Int: Int] = [
    0: 0,
    1: 1,
    2: 1
]

func solution(_ n: Int) -> Int {
    let answer = fibo(n)
    return answer % 1234567
}

func fibo(_ n: Int) -> Int {
    if dp[n] == nil {
        dp[n] = (fibo(n - 1) + fibo(n - 2)) % 1234567
    }
    return dp[n]!
}